src/sdk/namespace_blob.js

```diff
@@ -363,16 +341,40 @@ class NamespaceBlob {
     async create_object_upload(params, object_sdk) {
         // azure blob does not have start upload operation
         // instead starting multipart upload is implicit
-        // TODO how to handle xattr that s3 sends in the create command
-        const obj_id = crypto.randomBytes(24).toString('base64');
+        let obj_id = crypto.randomBytes(30).toString('base64');
```

Comment:
the entire handling of obj_id & etag in this namespace seems to be a bit shaky.
maybe we should refactor that handling to a function or two that will handle the generation of new ids for upload, the parsing of upload ids for multipart, and translation between existing blobs to s3 etag.

Also a very important thing - we need to make sure that etag that we return is either the real md5 of the data, or otherwise we the etag muat include a dash (`-`) because this is how S3 identifies etags of multipart uploads that are also not md5 of the assembled object. So the point is that returning etag=md5 of the object is the most compatible etag to AWS. After that return an etag with a dash which AWS SDK will identify and exclude from md5 checking. The dash option is less compatible than real md5, but if we don't intend on reading and computing the object md5 then the dash will make the etag more compatible than without one.

Reply:
This specific obj_id describes a multipartUpload uploadId - as a fix to this comment I returned the obj_id to be randomBytes(24) as it was before or to be the upload obj_id in a case where there are xattr, this fix simplify the changes in multipartUpload uploadId area.
also I added comments :)